### PR TITLE
feat: added appeal validation

### PR DESCRIPTION
**Pull request by ishanarya0**
add ValidateAppeal(a *domain.Appeal) method in provider service to validate the following things on appeal creation:

- [x] appeal options (access durations)
- [x] selected role

For BigQuery Provider:
- [x] validate permissions if they are valid/exist on bigquery provider creation

### CODE REVIEW

File: provider/metabase/config.go [hunk 1]
```diff
@@ -49,9 +49,7 @@ func (c *Credentials) Decrypt(decryptor domain.Decryptor) error {
 	return nil
 }
 
-type PermissionConfig struct {
-	Name string `json:"name" mapstructure:"name" validate:"required"`
-}
+type PermissionConfig string
```
Comment (rahmatrhd) [inline provider/metabase/config.go:52]:
```suggestion
type Permission string
```
Does this variable name sound better to you?

File: provider/service.go [hunk 2]
```diff
@@ -125,6 +125,40 @@ func (s *Service) GetRoles(id uint, resourceType string) ([]*domain.Role, error)
...
+	provider := s.getProvider(p.Type)
+	roles, err := provider.GetRoles(p.Config, resourceType)
+	if err != nil {
+		return err
+	}
+
+	isRoleExists := false
+	for _, role := range roles {
+		if a.Role == role.Name {
+			isRoleExists = true
```
Comment (rahmatrhd) [inline provider/service.go:143]:
```suggestion
			isRoleExists = true
			break
```

Comment (ishanarya0) [reply provider/service.go:143]: Added the break, thanks!

File: provider/metabase/config.go (hunk 1 above)
Comment (ishanarya0) [reply provider/metabase/config.go:52]: Yes, LGTM but if we keep PermissionConfig, then it would be consistent with all other providers. WDYT @rahmatrhd ?

Comment (rahmatrhd) [reply provider/metabase/config.go:52]: yes, let's rename all of them then if it is better

Comment (ishanarya0) [reply provider/metabase/config.go]: Sure! Done with the changes.

**Review by rahmatrhd — APPROVED** [review]

**Pull request merged**